### ISSUE TITLE
ASL | Transfer Learning and Fine-tuning for Sign Language

### ISSUE THREAD

**Issue by Varunshiyam** [UppuluriKalyani/ML-Nexus#675]
**Is your feature request related to a problem? Please describe.**

Effective communication with the deaf community is crucial for inclusivity, but traditional methods often present barriers.  This project aims to bridge this communication gap by developing a robust American Sign Language (ASL) recognition system using deep learning. The challenge lies in accurately classifying diverse ASL expressions from images to enable real-time translation and facilitate seamless communication.

-----

**Describe the solution you'd like**

This project demonstrates the application of transfer learning to classify American Sign Language (ASL) expressions. Two popular CNN architectures, InceptionV3 and ResNet50, are employed as base models with pre-trained weights from ImageNet.  The models are fine-tuned on an ASL dataset, leveraging their learned feature representations for improved accuracy.  The training process includes data augmentation and employs the Adam optimizer with a categorical cross-entropy loss function. The performance of both models is evaluated and compared, showcasing the effectiveness of transfer learning for ASL classification and its potential for communication support applications.

-----
**Kindly, Assign me this Issue Under:
- Gssoc🌸
- hacktoberfest☘️
- Level-3✨

------

**Additional context**

<img width="570" alt="Screenshot 2024-10-30 at 4 15 26 PM" src="https://github.com/user-attachments/assets/e51c34d6-c512-4eb1-8112-05da374e68c4">

<img width="581" alt="Screenshot 2024-10-30 at 4 15 48 PM" src="https://github.com/user-attachments/assets/46dd0e21-12e1-4b15-b603-6ac8dbc6407c">

<img width="569" alt="Screenshot 2024-10-30 at 4 16 23 PM" src="https://github.com/user-attachments/assets/d281c876-1d3f-4799-be39-6072d01bd487">

**Comment by Varunshiyam** [comment]
@Neilblaze @SaiNivedh26 Kindly Assign me this Issue under Level - 3 , Gssoc-ext , hacktoberfest

**Comment by UppuluriKalyani** [comment]
@Varunshiyam wait till I told you to do it 
And don't raise already existed issues it's already done btw.
